src/protocol/server.c

```diff
@@ -239,45 +268,75 @@ protocolServerProcess(
 }
 
 /**********************************************************************************************************************************/
-void
-protocolServerResponse(ProtocolServer *this, const Variant *output)
+PackRead *
+protocolServerDataGet(ProtocolServer *const this)
 {
     FUNCTION_LOG_BEGIN(logLevelTrace);
         FUNCTION_LOG_PARAM(PROTOCOL_SERVER, this);
-        FUNCTION_LOG_PARAM(VARIANT, output);
     FUNCTION_LOG_END();
 
-    KeyValue *result = kvNew();
+    PackRead *result = NULL;
 
-    if (output != NULL)
-        kvPut(result, VARSTR(PROTOCOL_OUTPUT_STR), output);
+    MEM_CONTEXT_TEMP_BEGIN()
+    {
+        PackRead *data = pckReadNew(this->read);
+        ProtocolMessageType type = (ProtocolMessageType)pckReadU32P(data);
 
-    ioWriteStrLine(protocolServerIoWrite(this), jsonFromKv(result));
-    ioWriteFlush(protocolServerIoWrite(this));
+        MEM_CONTEXT_PRIOR_BEGIN()
+        {
+            result = pckReadPackP(data);
+        }
+        MEM_CONTEXT_PRIOR_END();
 
-    FUNCTION_LOG_RETURN_VOID();
+        pckReadEndP(data);
+
+        CHECK(type == protocolMessageTypeData);
+    }
+    MEM_CONTEXT_TEMP_END();
+
+    FUNCTION_LOG_RETURN(PACK_READ, result);
 }
 
 /**********************************************************************************************************************************/
 void
-protocolServerWriteLine(ProtocolServer *this, const String *line)
+protocolServerDataPut(ProtocolServer *const this, PackWrite *const result)
```

Comment:
Why is this `result` here but in the protocol/server.h file the function signature has `resultPack`?

Reply:
Looks like I couldn't make up my mind! But really I think they should both be `data` like in `protocolClientDataPut()`.

Reply:
Fixed in f011c0e3a46.